Automattic/wp-calypso
# Framework: Bump ESLint dependencies to latest

This pull request seeks to update all of our ESLint dependencies to their latest available versions (with the exception of `babel-eslint`, upgraded to the version matching that defined in `eslint-plugin-wpcalypso` instead).

This brings a number of changes from our custom plugin and configuration, notably:

- Better validation of JSX className namespace (Automattic/eslint-plugin-wpcalypso#19)
- Enable [`template-curly-spacing`](http://eslint.org/docs/rules/template-curly-spacing) rule
- Removing an unintented debugging statement from the `wpcalypso/i18n-no-newlines` rule

See respective CHANGELOG for full details:

- https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/CHANGELOG.md
- https://github.com/Automattic/eslint-config-wpcalypso/blob/master/CHANGELOG.md
- https://github.com/eslint/eslint/blob/master/CHANGELOG.md

__Testing instructions:__

Verify that linting passes:

```
npm run lint
```

/cc @akirk @stuwest @deBhal @Copons 

Test live: https://calypso.live/?branch=update/eslint-latest
